### PR TITLE
Further development

### PR DESCRIPTION
**Pull request by mc12345678**
Some of the things to do was to continue on with the work last done, which included a little bit of javascript valiation as well as reuse of some of the code in other plugins that could take advantage of an image viewer.
Please look over these modifications for further incorporation.

### CODE REVIEW

**Review by Zen4All — APPROVED** [review]
I reviewed your updates, and as far as I can tell it all looks fine.
The extra options are an excellent idea.
What I never liked about this module ar all the jscript files in the different page folders. Is it an idea to move them into one file and make some conditional statements like "if ($current_page = product_info)" ?

**Comment by mc12345678** [comment]
I have mixed thoughts on the combining of the various page "specific" javascript. It makes changes to the group a lot easier, but it also reduces a sort of "new user" understanding/control. I mean, I didn't install all of the changes, but I would say that it was on purpose... of course if the test system stayed around and finally became a production system, I would likely forget why say colorbox wasn't working on a document_product page...

Basically thinking, such a change would need to not only be highlighted in the documentation as a difference but to also explain how to troubleshoot the issue because it would not be so clearly understood as being in the applicable includes/modules/pages/PROBLEM_PAGE folder. :) 

But, I guess as I just thought about this, a switch/case/default style would allow a "default" install and expansion by use of a notifier/observer if one so desired or they could go "old school" and place one of the versions of file in the applicable directory.

Dunno, bit of effort to modify, what issue(s) have been experienced by the way the code is currently applied in a page specific method? Is there some plan known about to work away from offering that capability and it would just be "better" for all for it to be so combined?
It would require a sort of cleanup to occur on next installation, ideally controlled/aided by way of the installer, or if absolutely necessary in two different directories such that the installer chooses to use one method or the other (nightmare for upkeep), etc...

**Comment by Zen4All** [comment]
Maybe for the sake of simplicity for end users we should keep the current file structure intact 😉 

For my own site I use the cs/js minifier, can't help being a fan of that addon. So I alway change my file locations and names for use with the minifier 😆 .

My proposal is to keep the structure the same as is for now, and maybe change it when Zen Cart 1.6 comes out, as structures wil be different in that version anyway.

**Comment by mc12345678** [comment]
So, I'm still a little behind on the ZC1.6.0 part of things because of the sweeping changes and how they are still evolving. But... if there is a "configuration" that is expected to not exist (ie. Many of the checkout_ related headers have been "worked out"), then I'd rather shove new/updated code into a condition that would continue to work in either/both ways sooner rather than later.

While I agree that the end user result when using the css/js loader is improved, there are other changes made by the plugin that I find unnecessary and annoying. Will install for those that ask/want, but is not something I use in whole.
So, I'm not against making a change to suit the bigger picture, just don't want to do so just because. Would rather less changes be necessary later if they can be supported now.

**Comment by Zen4All** [comment]
Don't get me wrong, it is absolutely not my intention to force the minifier upon other user. It is purely a personal taste. 😃

**Comment by mc12345678** [comment]
That was perhaps worth being said for others, but I didn't think that was the intent of the request/discussion.

I do find the placement of some of the files "odd".  Any suggestions/ideas about where the includes/classes/zen_colorbox related files really should reside? There are notes about using a similar "style" as was used for zen-lightbox, but just because two different plugins do things the same way doesn't make them both right... The code seems to act like modules, but also as javascript or perhaps javascript modules... Part of me wants to turn this more into a class style/related plugin but then such "modules" would be functions that return html... That has been a discussion/hardpoint for many.  

Trying to think/identify how things should go with ZC becoming more javascript/jquery/ajax centric other than the file structure of ajax style controlling functions being found in includes/classes/ajax.

**Comment by Zen4All** [comment]
In my opinion the whole folder can be moved the jscript in the template. All files are javascript/jquery related. They are definitely not class files.
